pandas/core/indexes/base.py

```diff
@@ -6960,14 +6959,8 @@ def _maybe_disable_logical_methods(self, opname: str_t) -> None:
         """
         raise if this Index subclass does not support any or all.
         """
-        if (
-            isinstance(self, ABCMultiIndex)
-            # TODO(3.0): PeriodArray and DatetimeArray any/all will raise,
-            #  so checking needs_i8_conversion will be unnecessary
-            or (needs_i8_conversion(self.dtype) and self.dtype.kind != "m")
-        ):
-            # This call will raise
-            make_invalid_op(opname)(self)
+        if isinstance(self, ABCMultiIndex) or self.dtype.kind != "m":
```

Comment:
is the != "m" bit still needed?

Reply:
Thanks for the comment. I think we don't need `!= "m"` here. This check worked only together with the check `needs_i8_conversion(self.dtype)`.

I removed `!= "m"` and corrected tests.

I have a question: shouldn't we raise for `DatetimeIndex`? So far we show  `FutureWarning`:
`"'any' with datetime64 dtypes is deprecated and will raise in a future version. Use (obj != pd.Timestamp(0)).any() instead."`

Reply:
there is a deprecation in nanany and nanall that i think is intended to be enforced before this needs_i8_conversion check is removed

Reply:
thanks, it was my thought too.
Should I open new PR to enforce the deprecation in `nanany` and `nanall`, or can I do it in this PR?

Reply:
I enforced deprecation `any/all` with `datetime64` in #58029